maths/three_sum.py

```diff
@@ -0,0 +1,45 @@
+def three_sum(nums: list[int]) -> list[list[int]]:
```

Comment:
```suggestion
def three_sum(nums: list[int]) -> tuple[tuple[int]]:
```

Reply:
I think list of list is better.

Reply:
Why?  Tuples take less memory and cannot be modified at runtime which might be useful characteristics.